### ISSUE TITLE
Synthetic events: unable to send a series of DELETE keys to a text input

### ISSUE THREAD

**Issue by lddubeau** [SeleniumHQ/selenium#1463]
### Versions

Selenium version: 2.48.0 all the way down to at least 2.45.0
Browser: Firefox 31 and 38, and probably other versions too...
OS: Linux

(Note: I see 2.48.2 as the latest release on github, but 2.48.0 is the latest Python package of Selenium available through pypi.)
### What steps will reproduce the problem?
1. This code reproduces the problem 100% of the time here:
   
   ```
   import selenium
   from selenium import webdriver
   from selenium.webdriver.firefox.webdriver import FirefoxBinary
   from selenium.webdriver.common.action_chains import ActionChains
   from selenium.webdriver.common.keys import Keys
   
   def perform(driver):
   driver.get("http://example.com")
   text = driver.execute_script("""
   var body = document.body;
   body.innerHTML += "<input id='text'>";
   var text = document.getElementById("text");
   text.value = "abcd";
   return text;
   """)
   
   print "selenium:", selenium.__version__
   print "browser:", driver.desired_capabilities["platform"], driver.name, \
       driver.desired_capabilities["version"]
   print "native events:", driver.desired_capabilities["nativeEvents"]
   print "initial value:", text.get_attribute("value")
   keys = [Keys.HOME] + [Keys.DELETE] * 4
   
   ActionChains(driver)\
       .click(text) \
       .send_keys(keys) \
       .perform()
   
   print "final value:", text.get_attribute("value")
   print
   
   driver.quit()
   
   driver = webdriver.Firefox(
   firefox_binary=FirefoxBinary("/home/ldd/src/firefox-31/firefox"),
   capabilities={"nativeEvents": True})
   perform(driver)
   
   driver = webdriver.Firefox(
   firefox_binary=FirefoxBinary("/home/ldd/src/firefox-31/firefox"),
   capabilities={"nativeEvents": False})
   perform(driver)
   
   driver = webdriver.Firefox(
   firefox_binary=FirefoxBinary("/home/ldd/src/firefox-38/firefox"))
   perform(driver)
   
   driver = webdriver.Chrome()
   perform(driver)
   ```
   
   Save it as `issue.py`. Edit the `FirefoxBinary` paths to point to your instances.
2. Run:
   
   python ./issue.py
### What is the expected output?

What the code does is click into a text input field and send HOME followed by 4 DELETE keystrokes. It should empty the text "abcd" that was pre-entered into the field.

In every single case of selenium version and browser version, the `final value:` line should show empty text because the field is empty.
### What do you see instead?

Only Chrome 47 (or any Chrome really) with any Selenium version, or Firefox 31.8 with Selenium 2.45.0 give the expected results.

In all other cases, we get `final value: bcd`, indicating that only one deletion happened.

Here is the output I get when running the test with Selenium 2.45.0:

```
selenium: 2.45.0
browser: Linux firefox 31.8.0
native events: True
initial value: abcd
final value:

selenium: 2.45.0
browser: Linux firefox 31.8.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.45.0
browser: Linux firefox 38.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.45.0
browser: Linux chrome 47.0.2526.106
native events: True
initial value: abcd
final value:
```

With Selenium 2.48.0:

```
selenium: 2.48.0
browser: Linux firefox 31.8.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.48.0
browser: Linux firefox 31.8.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.48.0
browser: Linux firefox 38.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.48.0
browser: Linux chrome 47.0.2526.106
native events: True
initial value: abcd
final value:
```
### Observations
1. The cases that produce incorrect results all correlate with the use of synthetic events in Firefox.
2. The `CHANGELOG` states that 2.48.0 "Supports native events for Firefox version 31 (immediately previous ESR)." but the test shows that even when native events are required with FF 31.8.0, Selenium 2.48.0 does not honor the request.

**Comment by barancev** [comment]
Is this a linux-specific issue?

Here is what I got with the latest versions of Selenium and Firefox:

selenium: 2.49.2
browser: WINDOWS firefox 43.0.4
native events: False
initial value: abcd
final value:

**Comment by HelioGuilherme66** [comment]
Still happens with ESR in Linux x64:
```selenium: 2.49.2
browser: Linux firefox 38.2.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.49.2
browser: Linux firefox 38.2.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.49.2
browser: Linux firefox 38.2.0
native events: False
initial value: abcd
final value: bcd

selenium: 2.49.2
browser: Linux chrome 48.0.2564.82
native events: True
initial value: abcd
final value: 

```

**Comment by lddubeau** [comment]
I can only reproduce the problem on Linux... and only if the Firefox window does not have focus. I did not initially realize that the presence or absence of focus changed anything. I've just discovered it now when rerunning the tests.

The root of the problem may be related to #825 since the issue there is also affected by focus.

For the record, I was able to reproduce the issue with Selenium 2.52.0 and FF 44.
